pkg/download/consistent_hashing.go

```diff
@@ -262,7 +263,28 @@ func (m *ConsistentHashingMode) DoRequest(ctx context.Context, start, end int64,
 
 	resp, err := m.Client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("error executing request for %s: %w", req.URL.String(), err)
+		if errors.Is(err, client.ErrStrategyFallback) {
```

Comment:
With a big caveat that I don't know the rest of this code very well, this looks like a surprising amount of code to implement a retry. Do we need to extract a function or a method here so that the code for making the initial request and the code to make the retry request only differ in a passed-in boolean `retry` (or an integer `retryCount` if you agree with my earlier suggestion)?

Reply:
I've done a refactor and removed this duplication.